### DISCUSSION TITLE
Alternative port with console proxy

### DISCUSSION THREAD

**Discussion by wagnerdracha** [apache/cloudstack#9713] [Q&A] [answered]
##### ISSUE TYPE

 * Other

##### COMPONENT NAME

Instances

##### CLOUDSTACK VERSION

CloudStack 4.19.1.0

##### OS / ENVIRONMENT

Ubuntu

##### SUMMARY

I access my cloudstack by https://example.com:8080/

I don't have all control of my modem ports. My ISP just give control of ports >= 1025. So, I need to do a port forward to access externally of my network.

When I login with cliente and try to access the proxy console, it try to connect with internal IP like 192.168.10.50. E.g.:
http://192.168.10.50/resource/noVNC/vnc.html?autoconnect=true&port=8080&token=[TOKEN]

Look here that the host address don't have the port 8080: *http://192.168.10.50*/....

So, how can I do a port forwarding to access enternally? What IP I have to point on port forwarding, management server?

And if I utilize a hostname (like *.consoleproxy.exemple.com.br), how can I add the port to this console proxy?

I saw that i have to update a configuration on global setting with the subdomain, so I created: *.consoleproxy.exemple.com.br

Now, when I acess the console proxy I got this url:
http://192.168.10.50.consoleproxy.exemple.com.br/resource/noVNC/vnc.html?autoconnect=true&port=8080&token=[TOKEN]
(something like that), cloudstack don´t access.

I didn't configure the SSL options.

How can I configure that option?

Thansk in advanced!

**Comment by weizhouapache** [comment, marked as answer]
the port 8080 is the internal port in CPVM (console proxy vm)  which used to communicate with vnc port or vnc over websocket connection. you do not need to consider it.

the open port of CPVM (console proxy vm) for vm consoles is hardcoded as 443 (https) or 80 (http). 
currently user can change the domain of console URL, but cannot change the port in the URL, you need to manually change the port in the URL  if the domain/public IP use a different port..
